pkg/registry/apis/folders/register.go

```diff
@@ -154,42 +158,60 @@ func (b *FolderAPIBuilder) PostProcessOpenAPI(oas *spec3.OpenAPI) (*spec3.OpenAP
 	return oas, nil
 }
 
+type authorizerParams struct {
+	user      identity.Requester
+	evaluator accesscontrol.Evaluator
+}
+
 func (b *FolderAPIBuilder) GetAuthorizer() authorizer.Authorizer {
-	return authorizer.AuthorizerFunc(
-		func(ctx context.Context, attr authorizer.Attributes) (authorized authorizer.Decision, reason string, err error) {
-			verb := attr.GetVerb()
-			name := attr.GetName()
-			if (!attr.IsResourceRequest()) || (name == "" && verb != utils.VerbCreate) {
-				return authorizer.DecisionNoOpinion, "", nil
-			}
+	return authorizer.AuthorizerFunc(func(ctx context.Context, attr authorizer.Attributes) (authorizer.Decision, string, error) {
```

Comment:
nit - do you really need an anonymous function here that calls your other function?  doesn't seem necessary

Reply:
I wanted to keep as much as the same signature as it was on main as the purpose of this PR is to be able to test getAuthorizer somehow. I decided to extract as much as I could from the already existing anonymous fn there. I agree in the future every getAuthorizer signature should be improved as it really prevents good unit testing.

Reply:
i think you could just create a function something like this
```
// function we can test
func myAuthorizer(ctx context.Context, attr authorizer.Attributes) (authorizer.Decision, string, error) {
/// ... your stuff here
}
```
then
```
return authorizer.AuthorizerFunc(myAuthorizer)
```
see this example:  https://go.dev/play/p/PLxyZ2o4Pz3